common/lib/client/auth.js

```diff
@@ -40,23 +40,31 @@ var Auth = (function() {
 
 	function Auth(client, options) {
 		this.client = client;
-		this.tokenParams = options.defaultTokenParams || {};
+		this._setOptions(options);
+	}
+
+	Auth.prototype._setOptions = function(options) {
+		this.tokenParams = this.tokenParams || options.defaultTokenParams || {};
 
-		/* RSA7a4: if options.clientId is provided and is not
-		 * null, it overrides defaultTokenParams.clientId */
 		if(options.clientId) {
```

Comment:
this should maybe be `if('clientId' in options)` -- if someone tries to pass in `{clientId: null}`, `""`, or `undefined`, setClientId should deal with that (whether by throwing an error or unsetting the clientId depending on whether we decide setOptions should be able to let you change your clientId)